lib/compilation_engine.rb

```diff
@@ -15,6 +17,9 @@ def compile_class
 
     output_token # class
     output_token # className
+
+    @output.puts("(class, defined, false, nil)")
```

Comment:
If I had used `#output_token` to emit the additional symbol table-related information. We can modify `#output_token` to receive additional arguments (i.e. the identifier category and index). We can also make `#output_token` use `@symbol_table` to define the identifiers.

Reply:
That could work, but adding arguments would mean we’d have to either add them to _all_ the calls of `#output_token` (they’re not relevant when we’re not outputting an identifier) or make them optional somehow. A more straightforward solution might be to make them part of the state of the `CompilationEngine`, i.e. store them in instance variables so that they’re available to all methods at all times, just like the current token is. I don’t know whether that actually makes sense here but it’s an alternative we could’ve tried if we’d wanted to.